olp-cpp-sdk-core/src/http/DefaultNetwork.h

```diff
@@ -55,14 +58,29 @@ class DefaultNetwork final : public Network {
   /// Implements the `SetDefaultHeaders` method of the `Network` class.
   void SetDefaultHeaders(Headers headers) override;
 
+  /// Implements the `SetCurrentBucket` method of the `Network` class.
+  void SetCurrentBucket(uint8_t bucket_id) override;
+
+  /// Implements the `GetStatistics` method of the `Network` class.
+  Statistics GetStatistics(uint8_t bucket_id) override;
+
  private:
   void AppendUserAgent(Headers& request_headers) const;
   void AppendDefaultHeaders(Headers& request_headers) const;
 
-  std::shared_ptr<Network> network_;
+  void LockStatistics(uint8_t bucket_id,
+                      std::function<void(Statistics&)>&& callback);
```

Comment:
Not sure if you need rvalue here. Probably pass by copy should also have the same effect if you only use it internally by reference.
```suggestion
                      std::function<void(Statistics&)> callback);
```